libraries/browser-tracker-core/src/tracker/out_queue.ts

```diff
@@ -230,14 +210,68 @@ export function OutQueueManager(
     return typeof queue[0] === 'object' && 'evt' in queue[0];
   };
 
+  // Runs `onRequestFailure` callback if defined, and clears the current batch
+  const tryOnRequestFailure = (status: number, message: string) => {
+    onRequestFailure?.({
+      events: currentBatch.slice(),
+      status,
+      message,
+    });
+    currentBatch.length = 0;
+  };
+
   /**
    * Send event as POST request right away without going to queue. Used when the request surpasses maxGetBytes or maxPostBytes
    * @param body POST request body
    * @param configCollectorUrl full collector URL with path
    */
   function sendPostRequestWithoutQueueing(body: PostEvent, configCollectorUrl: string) {
     const xhr = initializeXMLHttpRequest(configCollectorUrl, true, false);
-    xhr.send(encloseInPayloadDataEnvelope(attachStmToEvent([body.evt])));
+    const batch = attachStmToEvent([body.evt]);
+    currentBatch = batch as PostBatch;
+
+    // The timeout and onreadystatechange events are not set for oversized requests
+    // so we must set them here to listen for the response
+    setXhrCallbacks(xhr, 0);
```

Comment:
This changes a bit the behaviour from how we handled oversize requests before:

1. adds a timeout – this is good in theory but in practice it doesn't make much difference because oversize requests are not blocking other requests from being executed (`executingQueue` remains false while they are executed).
2. sets `executingQueue` to `false` after the request ends – I think we don't want to do this unless we set `executingQueue` to true also for oversize requests. This can have the side effect that in case there is another parallel request being made (which may happen), `executingQueue` is set to false prematurely.

I think we have two options:

1. handle oversize requests just like any other requests – block other requests from being made while they are executed (set `executingQueue` to true) and add timeout for them.
2. keep the behaviour as is and don't set `executingQueue` to `true` for oversize request, but also don't set it to `false` after they are done (also don't add timeout for them).

I think I prefer option 2 just for the reason that change is not related with this PR and if we want to change how oversize requests are done, we should do it separately.

Reply:
I've seperated out the `onreadystateupdate` callback for oversized payloads into its own thing which should be much more understandable now. All that happens now on completion is the firing of the callback for success/failure, if defined.